mozilla_version/test/test_maven.py

```diff
@@ -0,0 +1,82 @@
+import pytest
+
+from distutils.version import LooseVersion, StrictVersion
+
+from mozilla_version.errors import PatternNotMatchedError
+from mozilla_version.maven import MavenVersion
+
+@pytest.mark.parametrize('major_number, minor_number, patch_number, is_snapshot, expected_output_string', ((
+    32, 0, None, False, '32.0'
+), (
+    32, 0, 1, False, '32.0.1'
+), (
+    32, 0, None, True, '32.0-SNAPSHOT'
+), (
+    32, 0, 1, True, '32.0.1-SNAPSHOT'
+)))
+def test_maven_version_constructor_and_str(major_number, minor_number, patch_number, is_snapshot, expected_output_string):
+    assert str(MavenVersion(
+        major_number=major_number,
+        minor_number=minor_number,
+        patch_number=patch_number,
+        is_snapshot=is_snapshot,
+    )) == expected_output_string
+
+
+def test_maven_version_constructor_minimum_kwargs():
+    assert str(MavenVersion(32, 0)) == '32.0'
+    assert str(MavenVersion(32, 0, 1)) == '32.0.1'
+    assert str(MavenVersion(32, 1, 0)) == '32.1.0'
+    assert str(MavenVersion(32, 1, 0, False)) == '32.1.0'
+    assert str(MavenVersion(32, 1, 0, True)) == '32.1.0-SNAPSHOT'
+
+
+@pytest.mark.parametrize('version_string, ExpectedErrorType', (
+    ('32.0SNAPSHOT', PatternNotMatchedError),
+    ('32.1.0SNAPSHOT', PatternNotMatchedError),
+))
+def test_maven_version_raises_when_invalid_version_is_given(version_string, ExpectedErrorType):
+    with pytest.raises(ExpectedErrorType):
+        MavenVersion.parse(version_string)
+
+
+@pytest.mark.parametrize('previous, next', (
+    ('32.0-SNAPSHOT', '32.0'),
+    ('31.0', '32.0-SNAPSHOT'),
+    ('32.0', '32.0.1-SNAPSHOT'),
+    ('32.0.1-SNAPSHOT', '32.1.0'),
+))
```

Comment:
Should we insert a `('32.0.1-SNAPSHOT', '33.0')` just in case (very unlikely I know) there's a major release without SNAPSHOTS at all, following-up a SNAPSHOT dot release?

Reply:
Done.